### ISSUE TITLE
Bug in generation code

### ISSUE THREAD

**Issue by Squire-tomsk** [huggingface/parler-tts#85]
At this [line](https://github.com/huggingface/parler-tts/blob/8b8c576e2dbdc29172e30be7d68fac9357cd92c5/parler_tts/modeling_parler_tts.py#L1430), a delay pattern mask is generated and applied to the initial audio IDs. Then, at this [line](https://github.com/huggingface/parler-tts/blob/8b8c576e2dbdc29172e30be7d68fac9357cd92c5/parler_tts/modeling_parler_tts.py#L1530), a mask is also generated to revert the delay on the output tokens. However, it generates the mask based on `input_ids` which already includes the delay pattern.

**Comment by henry-tujia** [comment]
Here is just regenerating the mask once again to facilitate extracting the actual output of the model from the ids that already contain the mask.

**Comment by Squire-tomsk** [comment]
Yes, I understand the intention. However, the current code does not regenerate the mask in all cases. If the initial input_ids only contain a vector of BOS tokens, it works perfectly. But if you try to generate a continuation of some audio, the input_ids will be modified here [line](https://github.com/huggingface/parler-tts/blob/8b8c576e2dbdc29172e30be7d68fac9357cd92c5/parler_tts/modeling_parler_tts.py#L1430), and the mask generated here [line](https://github.com/huggingface/parler-tts/blob/8b8c576e2dbdc29172e30be7d68fac9357cd92c5/parler_tts/modeling_parler_tts.py#L1530) will be incorrect.

**Comment by Guppy16** [comment]
https://github.com/huggingface/parler-tts/pull/110

Does this help?
